policy-service/src/main/java/uk/gov/gchq/palisade/service/policy/PolicyApplication.java

```diff
@@ -16,26 +16,120 @@
+     */
+    public PolicyApplication(
+            final Collection<RunnableGraph<?>> runners,
+            final Materializer materializer,
+            final PolicyServiceCachingProxy service,
+            final PolicyConfiguration policyConfig,
+            final UserConfiguration userConfig,
+            final ResourceConfiguration resourceConfig,
+            @Qualifier("threadPoolTaskExecutor") final Executor executor) {
+        this.runners = new HashSet<>(runners);
+        this.materializer = materializer;
+        this.service = service;
+        this.policyConfig = policyConfig;
+        this.userConfig = userConfig;
+        this.resourceConfig = resourceConfig;
+        this.executor = executor;
+    }
+
+    /**
+     * Application entrypoint, creates and runs a spring application, passing in the given command-line args
+     *
+     * @param args command-line arguments passed to the application
+     */
     public static void main(final String[] args) {
-        LOGGER.debug("PolicyApplication started with: {} {} {}", PolicyApplication.class.toString(), "main", Arrays.toString(args));
+        LOGGER.debug("PolicyApplication started with: {}", (Object) args);
         new SpringApplicationBuilder(PolicyApplication.class).web(WebApplicationType.SERVLET)
                 .run(args);
     }
 
+    /**
+     * Runs all available Akka {@link RunnableGraph}s until completion.
+     * The 'main' threads of the application during runtime are the completable futures spawned here.
+     */
+    @EventListener(ApplicationReadyEvent.class)
+    public void serveForever() {
+        Set<CompletableFuture<?>> runnerThreads = runners.stream()
+                .map(runner -> CompletableFuture.supplyAsync(() -> runner.run(materializer), executor))
+                .collect(Collectors.toSet());
+        LOGGER.info("Started {} runner threads", runnerThreads.size());
+        runnerThreads.forEach(CompletableFuture::join);
+
+        //After launching Kafka, now run pre-population
+        LOGGER.debug("Pre-populating using policy config: {}", policyConfig.getClass());
+        LOGGER.debug("Pre-populating using user config: {}", userConfig.getClass());
+        LOGGER.debug("Pre-populating using resource config: {}", resourceConfig.getClass());
+        policyConfig.getPolicies().stream()
+                .forEach((PolicyPrepopulationFactory factory) -> {
+                    //Build Resource Rules
+                    Entry<Resource, Rules<LeafResource>> resourceMap = factory.buildResourceRules(resourceConfig.getResources());
+                    service.setResourceRules(resourceMap.getKey(), resourceMap.getValue());
+
+                    //Build Record Rules
+                    Entry<Resource, Rules<Serializable>> recordMap = factory.buildRecordRules(resourceConfig.getResources());
+                    service.setRecordRules(recordMap.getKey(), recordMap.getValue());
+                });
```

Comment:
These should be the other way around, prepopulate then start akka

Reply:
updated